minio/select_object_reader.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+#MinIO Python Library for Amazon S3 Compatible Cloud Storage, (C)
+# 2015, 2016, 2017, 2018, 2019 MinIO, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import io
+from PyCRC.CRC32 import CRC32
+from xml.etree import cElementTree
+from .error import InvalidXMLError
+from xml.etree.cElementTree import ParseError
+
+from .helpers import (READ_SIZE_SELECT, RECORDS,
+                      PROGRESS, STATS, EVENT, END, ERROR
+                      )
+
+
+class CRCValidationError(Exception):
+    '''
+    Raised in case of CRC mismatch
+    '''
+
+
+def calcuate_crc(value):
+    '''
+    Returns the CRC using PyCRC
+    '''
+    return CRC32().calculate(value)
+
+
+def validate_crc(current_value, expected_value):
+    '''
+    Validate through CRC check
+    '''
+    crc_current = calcuate_crc(current_value)
+    crc_expected = byte_int(expected_value)
+    if crc_current == crc_expected:
+        return True
+    return False
+
+
+def byte_int(data_bytes):
+    '''
+    Convert bytes to big-endian integer
+    '''
+    return int.from_bytes(data_bytes, byteorder='big')
+
+class SelectObjectReader(object): 
+    """
+    SelectObjectReader returns a Reader that upon read
+    returns queried data, but stops when the response ends.
+    LimitedRandomReader is compatible with BufferedIOBase.
+    """  
+    def __init__(self, remaining_bytes, response):
```

Comment:
We don't need to take remaining_bytes here, this seems to fail the current code as well.  The example is not working.

Reply:
corrected.